src/compile/legend/encode.ts

```diff
@@ -60,6 +60,9 @@ export function symbols(fieldDef: FieldDef<string>, symbolsSpec: any, model: Uni
       symbols.shape = {value: shapeDef.value};
     }
   }
+  if (channel !== OPACITY) {
+    symbols.opacity = model.encoding.opacity;
```

Comment:
It's weird that for shape we have to check if `model.encoding.shape` is a value def but here you don't check the same thing for opacity.

I don't know if the old code is wrong or your new code is wrong, but we definitely need to investigate what's right.  :)